scheduler/test/cook/test/mesos/task.clj

```diff
@@ -332,21 +333,40 @@
             framework-id {:value "framework-id"}
             environment {"COOK_JOB_CPUS" "1.0"
                          "COOK_JOB_MEM_MB" "10.0"
-                         "COOK_JOB_UUID" (-> job-ent :job/uuid str)}
-            task-metadata (task/job->task-metadata db framework-id executor job-ent task-id)]
-        (is (= {:command {:value "run-my-command", :environment environment, :user "test-user", :uris []}
-                :container nil
-                :environment environment
-                :executor :executor/custom
-                :executor-key :custom-executor
-                :framework-id framework-id
-                :labels {}
-                :name (format "dummy_job_%s_%s" (:job/user job-ent) task-id)
-                :num-ports 0
-                :resources {:cpus 1.0, :mem 10.0}
-                :task-id task-id}
-               (dissoc task-metadata :data)))
-        (is (= (pr-str {:instance "0"}) (-> task-metadata :data (String. "UTF-8"))))))
+                         "COOK_JOB_UUID" (-> job-ent :job/uuid str)}]
+
+        (testing "mesos-run-as-user absent"
+          (let [task-metadata (task/job->task-metadata db framework-id executor mesos-run-as-user job-ent task-id)]
+            (is (= {:command {:value "run-my-command", :environment environment, :user "test-user", :uris []}
+                    :container nil
+                    :environment environment
+                    :executor :executor/custom
+                    :executor-key :custom-executor
+                    :framework-id framework-id
+                    :labels {}
+                    :name (format "dummy_job_%s_%s" (:job/user job-ent) task-id)
+                    :num-ports 0
+                    :resources {:cpus 1.0, :mem 10.0}
+                    :task-id task-id}
+                   (dissoc task-metadata :data)))
+            (is (= (pr-str {:instance "0"}) (-> task-metadata :data (String. "UTF-8"))))))
+
+        (testing "mesos-run-as-user present"
+          (let [mesos-run-as-user "mesos-run-as-user"
+                task-metadata (task/job->task-metadata db framework-id executor mesos-run-as-user job-ent task-id)]
+            (is (= {:command {:value "run-my-command", :environment environment, :user mesos-run-as-user, :uris []}
+                    :container nil
+                    :environment environment
+                    :executor :executor/custom
+                    :executor-key :custom-executor
+                    :framework-id framework-id
+                    :labels {}
+                    :name (format "dummy_job_%s_%s" (:job/user job-ent) task-id)
+                    :num-ports 0
+                    :resources {:cpus 1.0, :mem 10.0}
+                    :task-id task-id}
+                   (dissoc task-metadata :data)))
+            (is (= (pr-str {:instance "0"}) (-> task-metadata :data (String. "UTF-8"))))))))
```

Comment:
For both of these tests, how about if we only assert on the `:command` `:user`, since that's what we are interested in here?

Reply:
Since it is a different invocation, I assert on the entire output in both versions of the tests.

Reply:
We talked offline, agree to disagree :)